internal/pkg/docker/dockercompose/decompose.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockercompose
+
+import (
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/compose-spec/compose-go/loader"
+	compose "github.com/compose-spec/compose-go/types"
+	"sort"
+)
+
+// DecomposeService parses a Compose YAML file and then converts a single service to a Copilot manifest.
+func DecomposeService(content []byte, svcName string, workingDir string) (*manifest.BackendServiceConfig, IgnoredKeys, error) {
+	config, err := loader.ParseYAML(content)
+	if err != nil {
+		return nil, nil, fmt.Errorf("parse compose yaml: %w", err)
+	}
+
+	service, err := serviceConfig(config, svcName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get service config: %w", err)
+	}
+
+	ignored, err := unsupportedServiceKeys(service, svcName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get unsupported service keys: %w", err)
+	}
+
+	project, err := loader.Load(compose.ConfigDetails{
+		WorkingDir: workingDir,
+		ConfigFiles: []compose.ConfigFile{
+			{
+				Content: content,
+				Config:  config,
+			},
+		},
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("load Compose project: %w", err)
+	}
+
+	svcConfig, err := project.GetService(svcName)
+	if err != nil {
+		// I don't think this is possible due to the previous check,
+		// but I'm not confident enough in that assessment to make this a panic.
+		return nil, nil, fmt.Errorf("get service from Compose project: %w", err)
+	}
+
+	// TODO: Port handling & exposed port detection, to be implemented in Milestone 3
+	var port uint16 = 80
+	backendSvc, svcIgnored, err := convertBackendService(&svcConfig, port)
+	if err != nil {
+		return nil, nil, fmt.Errorf("convert Compose service to Copilot manifest: %w", err)
+	}
+
+	ignored = append(ignored, svcIgnored...)
+	sort.Strings(ignored)
+
+	return backendSvc, ignored, nil
+}
+
+// serviceConfig extracts the map corresponding to a Compose service from the parsed Compose config map.
+func serviceConfig(config map[string]interface{}, svcName string) (map[string]interface{}, error) {
+	svcs, ok := config["services"]
+	if !ok || svcs == nil {
```

Comment:
If svcs is a map then i would recommend
```suggestion
	if !ok || len(svcs) == 0 {
```

Reply:
Unfortunately, I have to revert this, since `svcs` is of type `interface{}` instead of being a map.

Reply:
gotcha that makes sense.